### DISCUSSION TITLE
Advanced search for a specific document name in a repository

### DISCUSSION THREAD

**Discussion by jeb2260** [community/community#22877] [New to GitHub]
<p>Hi,</p>
<p>If I would like to search for all repositories created since yesterday that have the document “whitepaper.pdf” contained in the repository. How would I do this? </p>
<p>I already took a look at the <a href="https://help.github.com/articles/searching-repositories/" rel="nofollow noopener">Searching Repositories</a> page but am still unsure. Any help would be great!</p>

**Comment by xarg** [comment]
<p>You can hit the <code>t</code> key on your keyboard (while on the repository page) and then search for the file you need.</p>

**Comment by jeb2260** [comment]
<p>I’m looking to search for all repositories created past the certain date that have the specified file in it, so I wouldn’t have to manually check each repository</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t a way to do what you want from the UI in one step. You could use the <a href="https://developer.github.com/v3/search/" rel="nofollow noopener">Search API</a> to write a script that would first do the repository search and then search the matching repositories for the file you were looking for.</p>
<p>I hope that helps!</p>
